stjude/proteinpaint
# Init routes

## Description

This PR:
- moves most of the route API definition from server/routes to shared/types/src/routes
- makes route and type def names more consistent
- restores the autogeneration of server route documentation and browser testing (command line testing will be restored in a separate PR)

To test:
- http://localhost:3000/testrun.html?name=*.unit
- `cd client && npm run test:integration && npm run tsc`
- `cd ../server && npx tsc`
- `cd ../shared/types && npx tsc`
- `cd sjpp/proteinpaint && npm run doc`, then restart `npm run dev`, and http://localhost:3000/server.html should load routes

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
